core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -82,19 +82,57 @@ class GroupMetadataManager(brokerId: Int,
 
   this.logIdent = s"[GroupMetadataManager brokerId=$brokerId] "
 
-  newGauge("NumOffsets",
+  private def recreateGauge[T](name: String, gauge: Gauge[T]): Gauge[T] = {
+    removeMetric(name)
+    newGauge(name, gauge)
+  }
+
+  recreateGauge("NumOffsets",
     new Gauge[Int] {
       def value = groupMetadataCache.values.map(group => {
         group synchronized { group.numOffsets }
       }).sum
-    }
-  )
+    })
 
-  newGauge("NumGroups",
+  val numGroupsGauge = recreateGauge("NumGroups",
```

Comment:
I'm not sure these need to be fields. It's a little more work, but we can pull the metric instances out of the metric registry in the test case.

Also, should we update the test case to cover all the metrics?

Reply:
Yeah, other tests usually just retrieve the data from the metrics registry as you said.

Reply:
OK...